project/tests/com/google/daggerquery/executor/models/QueryTest.java

```diff
@@ -107,6 +109,97 @@ public void testExecutingDepsQuery_WithNullBindingGraph_ThrowsNullPointerExcepti
     query.execute(null);
   }
 
+  // Tests for `ALLPATHS` query
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testParsingAllPathsQuery_WithOneStringParameter_ThrowsIllegalArgumentException() {
+    Query query = new Query("allpaths", "com.google.cats.Cat");
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testParsingAllPathsQuery_WithoutParameters_ThrowsIllegalArgumentException() {
+    Query query = new Query("allpaths");
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testParsingAllPathsQuery_WithNullParameters_ThrowsNullPointerException() {
+    Query query = new Query("allpaths", /*parameters = */ null);
+  }
+
+  @Test
+  public void testExecutingAllPathsQuery_WithCapitalizedName_WhenResultContainsOnePathWithTwoNodes() {
+    String[] parameters = {"com.google.Cat", "com.google.Details"};
+    Query query = new Query("ALLpaths", parameters);
+
+    List<String> queryExecutionResult = query.execute(makeBindingGraph_WithMultiplePathsBetweenTwoNodes());
+
+    String[] expectedOutput = {"com.google.Cat -> com.google.Details"};
+    assertArrayEquals(expectedOutput, queryExecutionResult.toArray());
+  }
+
+  @Test
+  public void testExecutingAllPathsQuery_WhenResultContainsOnePathWithThreeNodes() {
+    String[] parameters = {"com.google.CatsFactory", "com.google.Details"};
+    Query query = new Query("allpaths", parameters);
+
+    List<String> queryExecutionResult = query.execute(makeBindingGraph_WithMultiplePathsBetweenTwoNodes());
+
+    String[] expectedOutput = {"com.google.CatsFactory -> com.google.Cat -> com.google.Details"};
+    assertArrayEquals(expectedOutput, queryExecutionResult.toArray());
+  }
+
+  @Test
+  public void testExecutingAllPathsQuery_WhenResultContainsMultiplePaths() {
+    String[] parameters = {"com.google.Component", "com.google.Details"};
+    Query query = new Query("allpaths", parameters);
+
+    List<String> queryExecutionResult = query.execute(makeBindingGraph_WithMultiplePathsBetweenTwoNodes());
+
+    String[] expectedOutput = {
+        "com.google.Component -> com.google.CatsFactory -> com.google.Cat -> com.google.Details",
+        "com.google.Component -> com.google.Helper -> com.google.CatsFactory -> com.google.Cat -> com.google.Details",
+        "com.google.Component -> com.google.Cat -> com.google.Details",
+        "com.google.Component -> com.google.Details"
+    };
```

Comment:
Can we guarantee that the ordering of these Strings be deterministic? 
(AKA that they will be in the same order each time)?

I think this can depend on a number of things:
1) How dagger constructs it's binding graph.
2) How dagger plugins traverse the binding graph.
3) How protobufs store data. 
4) If the algorithm for the query executor is deterministic.

Reply:
I've [changed the test](https://github.com/googleinterns/dagger-query/pull/15/commits/85d91497ef42eb14ebf1fdc7fd1faa79c90d0320) to ignore an ordering because it doesn't matter in which order paths are listed so this one-to-one comparison is not necessary.